### PR TITLE
feat: kv additions

### PR DESCRIPTION
**Pull request by ashleymichal**
Long-running feature branch for KV Additions Milestone (https://github.com/cloudflare/wrangler/milestone/10).

Comment-only reviews until WIP removed please.

### CODE REVIEW

File: src/commands/kv/bucket/delete.rs [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    namespace_id: &str,
+    filename: &Path,
+) -> Result<(), failure::Error> {
+    let keys: Result<Vec<String>, failure::Error> = match &metadata(filename) {
+        Ok(file_type) if file_type.is_dir() => directory_keys_only(filename),
+        Ok(_) => failure::bail!("{} should be a directory", filename.display()),
+        Err(e) => failure::bail!("{}", e),
+    };
+
+    delete_bulk(target, user, namespace_id, keys?)
```
Comment (ashleymichal) [inline src/commands/kv/bucket/delete.rs]: does this command now need to give output?

File: src/commands/kv/bucket/upload.rs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    let pairs: Result<Vec<KeyValuePair>, failure::Error> = match &metadata(filename) {
+        Ok(file_type) if file_type.is_dir() => directory_keys_values(filename),
+        Ok(_file_type) => {
+            // any other file types (files, symlinks)
+            failure::bail!("wrangler kv:bucket upload takes a directory")
+        }
+        Err(e) => failure::bail!("{}", e),
+    };
+
+    put_bulk(target, user, namespace_id, pairs?)
```
Comment (ashleymichal) [inline src/commands/kv/bucket/upload.rs]: same question here

File: src/commands/kv/bucket/delete.rs (hunk 1 above)
Comment (gabbifish) [reply src/commands/kv/bucket/delete.rs]: It does!

File: src/commands/kv/bucket/upload.rs (hunk 2 above)
Comment (gabbifish) [reply src/commands/kv/bucket/upload.rs]: It does :)

File: src/commands/kv/bucket/delete.rs (hunk 1 above)
Comment (gabbifish) [reply src/commands/kv/bucket/delete.rs]: this is handled in #601

File: src/commands/kv/bucket/upload.rs (hunk 2 above)
Comment (gabbifish) [reply src/commands/kv/bucket/upload.rs]: this is handled in #601

**Pull request merged**